### PR TITLE
gpy: unbreak on python3.9

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
Gpy is currently not compiling on python3.9 (https://hydra.nixos.org/build/133670737)

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by prusnak** [comment]
@GrahamcOfBorg build python37Packages.gpy python38Packages.gpy python39Packages.gpy

File: pkgs/development/python-modules/gpy/default.nix [hunk 1]
```diff
@@ -19,6 +19,14 @@ buildPythonPackage rec {
...
   buildInputs = [ cython ];
 
+  # Force cython-generated .c files to be re-built since the included
+  # ones were built with an older version of cython that produces .c
+  # files which are incompatible with python3.9
+  preConfigure = ''
+    echo "[build_ext]" >> setup.cfg
+    echo "force=1" >> setup.cfg
+  '';
+
```
Comment (prusnak) [inline pkgs/development/python-modules/gpy/default.nix:29]: It seems other modules are using the following? Is this the same functionality?
```
  preBuild = ''
    ${python.interpreter} setup.py build_ext
  '';
```

Comment (rmcgibbo) [reply pkgs/development/python-modules/gpy/default.nix:29]: Just tried it -- apparently no. I do think another tactic that would work is just removing the generated .c files, which would force cython to re-translate them.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107653` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.gpy</li>
    <li>python37Packages.gpyopt</li>
    <li>python38Packages.gpy</li>
    <li>python38Packages.gpyopt</li>
    <li>python39Packages.gpy</li>
    <li>python39Packages.gpyopt</li>
  </ul>
</details>

File: pkgs/development/python-modules/gpy/default.nix (hunk 1 above)
Comment (prusnak) [reply pkgs/development/python-modules/gpy/default.nix:29]: My two cents: removing pre-generated *.c files in `preBuild` is probably easier to understand, but I am fine with the proposed change too.

Comment (rmcgibbo) [reply pkgs/development/python-modules/gpy/default.nix]: Agreed. Simply removing them didn't do it though, but I force-pushed a new version that just regenerates the cython files ourselves in preBuild.

**Comment by SuperSandro2000** [comment]
Nixpkgs-review loops on darwin forever.

Also please change the titel and commit message to ``python39Packages.gpy: unbreak on python3.9`` or similar.

**Comment by rmcgibbo** [comment]
@SuperSandro2000: Done.

I'll dust off my mac to test. I wonder why it hangs.

**Comment by SuperSandro2000** [comment]
> I'll dust off my mac to test. I wonder why it hangs.

I don't fully understand it and which information is useful because it logged several 10k lines.

**Comment by rmcgibbo** [comment]
@SuperSandro2000: Can reproduce. I'll debug this tomorrow.

**Comment by rmcgibbo** [comment]
For some reason, it was the build of one of the upstream dependencies (paramz) that was looping on darwin. I appear to have fixed that now (although I don't understand what the cause was).

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107653` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python37Packages.gpy</li>
    <li>python37Packages.gpyopt</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>python37Packages.paramz</li>
    <li>python38Packages.gpy</li>
    <li>python38Packages.gpyopt</li>
    <li>python38Packages.paramz</li>
    <li>python39Packages.gpy</li>
    <li>python39Packages.gpyopt</li>
    <li>python39Packages.paramz</li>
  </ul>
</details>

python37Packages.gpy
```
running install tests
...................................F........... /nix/store/c7y36xrsgm31phhi786sshksj47pwll6-python3.7-paramz-0.9.5/lib/python3.7/site-packages/paramz/transformations.py:111: RuntimeWarnin>... /build/GPy-1.9.9/GPy/core/gp.py:87: UserWarning:Your kernel has a different input dimension 1 then the given X dimension 2. Be very sure this is what you want and you have not forgott>............... /build/GPy-1.9.9/GPy/core/gp.py:87: UserWarning:Your kernel has a different input dimension 4 then the given X dimension 5. Be very sure this is what you want and you have>............... /build/GPy-1.9.9/GPy/util/warping_functions.py:142: RuntimeWarning:overflow encountered in cosh
 /build/GPy-1.9.9/GPy/util/warping_functions.py:143: RuntimeWarning:overflow encountered in cosh
 /build/GPy-1.9.9/GPy/util/warping_functions.py:144: RuntimeWarning:overflow encountered in cosh
 /build/GPy-1.9.9/GPy/util/warping_functions.py:152: RuntimeWarning:overflow encountered in cosh
 /build/GPy-1.9.9/GPy/util/warping_functions.py:153: RuntimeWarning:overflow encountered in cosh
...
======================================================================
FAIL: Testing the TP regression with matern32 kernel on 1d data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/GPy-1.9.9/GPy/testing/model_tests.py", line 843, in test_TPRegression_matern32_1D
    self.check_model(matern32, model_type='TPRegression', dimension=1)
  File "/build/GPy-1.9.9/GPy/testing/model_tests.py", line 682, in check_model
    self.assertTrue(m.checkgrad())
AssertionError: False is not true

----------------------------------------------------------------------
Ran 83 tests in 239.213s

FAILED (failures=1)
```

**Comment by rmcgibbo** [comment]
Might be a flaky test -- it worked for me on linux x64. But `239.213s` is a bit too long anyways, so I'll disable some more tests.

```
[mcgibbon@pn50:~/projects/nixpkgs/pkgs/development/python-modules]$ nixpkgs-review pr 107653
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/107653/head:refs/nixpkgs-review/1
remote: Enumerating objects: 393, done.
remote: Counting objects: 100% (393/393), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 587 (delta 388), reused 388 (delta 388), pack-reused 194
Receiving objects: 100% (587/587), 472.80 KiB | 5.20 MiB/s, done.
Resolving deltas: 100% (423/423), completed with 160 local objects.
From https://github.com/NixOS/nixpkgs
   9ed0fa8cf38..05140d25dcb  master                -> refs/nixpkgs-review/0
 + 0b4e4bcb055...e4fb207180c refs/pull/107653/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/mcgibbon/.cache/nixpkgs-review/pr-107653/nixpkgs 05140d25dcb5f491613b0c76bd39eecd97241948
Preparing worktree (detached HEAD 05140d25dcb)
HEAD is now at 05140d25dcb Merge pull request #105462 from ruuda/mypy-mypyc
$ git merge --no-commit e4fb207180c7d1442f85c4c441bc694a00c23d7d
Automatic merge went well; stopped before committing as requested
$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/mcgibbon/.cache/nixpkgs-review/pr-107653/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/107653

9 packages built:
python37Packages.gpy python37Packages.gpyopt python37Packages.paramz python38Packages.gpy python38Packages.gpyopt python38Packages.paramz python39Packages.gpy python39Packages.gpyopt python39Packages.paramz
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107653` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python37Packages.gpy</li>
    <li>python37Packages.gpyopt</li>
    <li>python37Packages.paramz</li>
    <li>python38Packages.gpy</li>
    <li>python38Packages.gpyopt</li>
    <li>python38Packages.paramz</li>
    <li>python39Packages.gpy</li>
    <li>python39Packages.gpyopt</li>
    <li>python39Packages.paramz</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107653` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python37Packages.gpy</li>
    <li>python37Packages.gpyopt</li>
    <li>python38Packages.gpy</li>
    <li>python38Packages.gpyopt</li>
    <li>python39Packages.gpy</li>
    <li>python39Packages.gpyopt</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.paramz</li>
    <li>python38Packages.paramz</li>
    <li>python39Packages.paramz</li>
  </ul>
</details>
```
running install tests
..........................................F.... /nix/store/v2hbg74iv430p0s6qjlfifr3qfnx48az-python3.7-paramz-0.9.5/lib/python3.7/site-packages/paramz/transformations.py:111: RuntimeWarning:overflow encountered in expm1
... /private/tmp/nix-build-python3.7-GPy-1.9.9.drv-0/GPy-1.9.9/GPy/core/gp.py:87: UserWarning:Your kernel has a different input dimension 1 then the given X dimension 2. Be very sure this is what you want and you have not forgotten to set the right input dimenion in your kernel
............... /private/tmp/nix-build-python3.7-GPy-1.9.9.drv-0/GPy-1.9.9/GPy/core/gp.py:87: UserWarning:Your kernel has a different input dimension 4 then the given X dimension 5. Be very sure this is what you want and you have not forgotten to set the right input dimenion in your kernel
............... /private/tmp/nix-build-python3.7-GPy-1.9.9.drv-0/GPy-1.9.9/GPy/util/warping_functions.py:142: RuntimeWarning:overflow encountered in cosh
 /private/tmp/nix-build-python3.7-GPy-1.9.9.drv-0/GPy-1.9.9/GPy/util/warping_functions.py:143: RuntimeWarning:overflow encountered in cosh
 /private/tmp/nix-build-python3.7-GPy-1.9.9.drv-0/GPy-1.9.9/GPy/util/warping_functions.py:144: RuntimeWarning:overflow encountered in cosh
 /private/tmp/nix-build-python3.7-GPy-1.9.9.drv-0/GPy-1.9.9/GPy/util/warping_functions.py:152: RuntimeWarning:overflow encountered in cosh
 /private/tmp/nix-build-python3.7-GPy-1.9.9.drv-0/GPy-1.9.9/GPy/util/warping_functions.py:153: RuntimeWarning:overflow encountered in cosh
2020-12-27 23:14:04.684 python3.7[73656:16254334] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Critical error: required built-in appearance SystemAppearance not found'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff32e82727 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff6bd8aa9e objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff32e82585 +[NSException raise:format:] + 181
	3   libdispatch.dylib                   0x00007fff6ced1658 _dispatch_client_callout + 8
	4   libdispatch.dylib                   0x00007fff6ced27de _dispatch_once_callout + 20
	5   AppKit                              0x00007fff3002bfda +[NSAppearance _initializeCoreUI] + 35
	6   AppKit                              0x00007fff3005137d +[NSAppearance _aquaAppearance] + 51
	7   AppKit                              0x00007fff3002b964 +[NSAppearance appearanceNamed:] + 21
	8   AppKit                              0x00007fff3002b6b3 _NSAppearanceCurrentSystemAppearance + 178
	9   AppKit                              0x00007fff3002b486 -[NSSystemAppearanceProxy init] + 81
	10  AppKit                              0x00007fff3002b42c __38+[NSSystemAppearanceProxy systemProxy]_block_invoke + 24
	11  libdispatch.dylib                   0x00007fff6ced1658 _dispatch_client_callout + 8
	12  libdispatch.dylib                   0x00007fff6ced27de _dispatch_once_callout + 20
	13  AppKit                              0x00007fff3002b412 +[NSSystemAppearanceProxy systemProxy] + 42
	14  AppKit                              0x00007fff3002b3be -[NSApplication(NSApplicationAppearance_Internal) _registerForAppearanceNotifications] + 34
	15  AppKit                              0x00007fff30028ee5 -[NSApplication init] + 669
	16  AppKit                              0x00007fff300289e9 +[NSApplication sharedApplication] + 123
	17  _macosx.cpython-37m-darwin.so       0x000000010c9e9f89 FigureCanvas_new + 57
	18  libpython3.7m.dylib                 0x0000000107a6270c type_call + 44
	19  libpython3.7m.dylib                 0x0000000107a0ed21 _PyObject_FastCallKeywords + 577
	20  libpython3.7m.dylib                 0x0000000107aeb435 call_function + 789
	21  libpython3.7m.dylib                 0x0000000107ae7f83 _PyEval_EvalFrameDefault + 29171
	22  libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	23  libpython3.7m.dylib                 0x0000000107aeb449 call_function + 809
	24  libpython3.7m.dylib                 0x0000000107ae7f83 _PyEval_EvalFrameDefault + 29171
	25  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	26  libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	27  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	28  libpython3.7m.dylib                 0x0000000107a0f028 PyObject_Call + 136
	29  libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	30  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	31  libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	32  libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	33  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	34  libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	35  libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	36  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	37  libpython3.7m.dylib                 0x0000000107a0eec6 _PyFunction_FastCallKeywords + 230
	38  libpython3.7m.dylib                 0x0000000107aeb449 call_function + 809
	39  libpython3.7m.dylib                 0x0000000107ae7f63 _PyEval_EvalFrameDefault + 29139
	40  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	41  libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	42  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	43  libpython3.7m.dylib                 0x0000000107a0f028 PyObject_Call + 136
	44  libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	45  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	46  libpython3.7m.dylib                 0x0000000107a0eec6 _PyFunction_FastCallKeywords + 230
	47  libpython3.7m.dylib                 0x0000000107aeb449 call_function + 809
	48  libpython3.7m.dylib                 0x0000000107ae7f63 _PyEval_EvalFrameDefault + 29139
	49  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	50  libpython3.7m.dylib                 0x0000000107a0eec6 _PyFunction_FastCallKeywords + 230
	51  libpython3.7m.dylib                 0x0000000107aeb449 call_function + 809
	52  libpython3.7m.dylib                 0x0000000107ae802e _PyEval_EvalFrameDefault + 29342
	53  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	54  libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	55  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	56  libpython3.7m.dylib                 0x0000000107a0f028 PyObject_Call + 136
	57  libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	58  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	59  libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	60  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	61  libpython3.7m.dylib                 0x0000000107a65d2b slot_tp_call + 171
	62  libpython3.7m.dylib                 0x0000000107a0ed21 _PyObject_FastCallKeywords + 577
	63  libpython3.7m.dylib                 0x0000000107aeb435 call_function + 789
	64  libpython3.7m.dylib                 0x0000000107ae802e _PyEval_EvalFrameDefault + 29342
	65  libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	66  libpython3.7m.dylib                 0x0000000107aeb449 call_function + 809
	67  libpython3.7m.dylib                 0x0000000107ae7f63 _PyEval_EvalFrameDefault + 29139
	68  libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	69  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	70  libpython3.7m.dylib                 0x0000000107a0f028 PyObject_Call + 136
	71  libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	72  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	73  libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	74  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	75  libpython3.7m.dylib                 0x0000000107a65d2b slot_tp_call + 171
	76  libpython3.7m.dylib                 0x0000000107a0ed21 _PyObject_FastCallKeywords + 577
	77  libpython3.7m.dylib                 0x0000000107aeb435 call_function + 789
	78  libpython3.7m.dylib                 0x0000000107ae802e _PyEval_EvalFrameDefault + 29342
	79  libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	80  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	81  libpython3.7m.dylib                 0x0000000107a0f028 PyObject_Call + 136
	82  libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	83  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	84  libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	85  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	86  libpython3.7m.dylib                 0x0000000107a65d2b slot_tp_call + 171
	87  libpython3.7m.dylib                 0x0000000107a0ed21 _PyObject_FastCallKeywords + 577
	88  libpython3.7m.dylib                 0x0000000107aeb435 call_function + 789
	89  libpython3.7m.dylib                 0x0000000107ae802e _PyEval_EvalFrameDefault + 29342
	90  libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	91  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	92  libpython3.7m.dylib                 0x0000000107a0f028 PyObject_Call + 136
	93  libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	94  libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	95  libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	96  libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	97  libpython3.7m.dylib                 0x0000000107a65d2b slot_tp_call + 171
	98  libpython3.7m.dylib                 0x0000000107a0ed21 _PyObject_FastCallKeywords + 577
	99  libpython3.7m.dylib                 0x0000000107aeb435 call_function + 789
	100 libpython3.7m.dylib                 0x0000000107ae802e _PyEval_EvalFrameDefault + 29342
	101 libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	102 libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	103 libpython3.7m.dylib                 0x0000000107a0f028 PyObject_Call + 136
	104 libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	105 libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	106 libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	107 libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	108 libpython3.7m.dylib                 0x0000000107a65d2b slot_tp_call + 171
	109 libpython3.7m.dylib                 0x0000000107a0ed21 _PyObject_FastCallKeywords + 577
	110 libpython3.7m.dylib                 0x0000000107aeb435 call_function + 789
	111 libpython3.7m.dylib                 0x0000000107ae802e _PyEval_EvalFrameDefault + 29342
	112 libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	113 libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	114 libpython3.7m.dylib                 0x0000000107a0f028 PyObject_Call + 136
	115 libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	116 libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	117 libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	118 libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	119 libpython3.7m.dylib                 0x0000000107a65d2b slot_tp_call + 171
	120 libpython3.7m.dylib                 0x0000000107a0ed21 _PyObject_FastCallKeywords + 577
	121 libpython3.7m.dylib                 0x0000000107aeb435 call_function + 789
	122 libpython3.7m.dylib                 0x0000000107ae802e _PyEval_EvalFrameDefault + 29342
	123 libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	124 libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	125 libpython3.7m.dylib                 0x0000000107a0f028 PyObject_Call + 136
	126 libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	127 libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	128 libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	129 libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	130 libpython3.7m.dylib                 0x0000000107a65d2b slot_tp_call + 171
	131 libpython3.7m.dylib                 0x0000000107a0ed21 _PyObject_FastCallKeywords + 577
	132 libpython3.7m.dylib                 0x0000000107aeb435 call_function + 789
	133 libpython3.7m.dylib                 0x0000000107ae802e _PyEval_EvalFrameDefault + 29342
	134 libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	135 libpython3.7m.dylib                 0x0000000107aeb449 call_function + 809
	136 libpython3.7m.dylib                 0x0000000107ae7f63 _PyEval_EvalFrameDefault + 29139
	137 libpython3.7m.dylib                 0x0000000107a0f3fd function_code_fastcall + 253
	138 libpython3.7m.dylib                 0x0000000107aeb449 call_function + 809
	139 libpython3.7m.dylib                 0x0000000107ae7f63 _PyEval_EvalFrameDefault + 29139
	140 libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	141 libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	142 libpython3.7m.dylib                 0x0000000107ae829f _PyEval_EvalFrameDefault + 29967
	143 libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	144 libpython3.7m.dylib                 0x0000000107a0e995 _PyFunction_FastCallDict + 501
	145 libpython3.7m.dylib                 0x0000000107a0ffb6 _PyObject_Call_Prepend + 150
	146 libpython3.7m.dylib                 0x0000000107a67311 slot_tp_init + 177
	147 libpython3.7m.dylib                 0x0000000107a62804 type_call + 292
	148 libpython3.7m.dylib                 0x0000000107a0ed21 _PyObject_FastCallKeywords + 577
	149 libpython3.7m.dylib                 0x0000000107aeb435 call_function + 789
	150 libpython3.7m.dylib                 0x0000000107ae802e _PyEval_EvalFrameDefault + 29342
	151 libpython3.7m.dylib                 0x0000000107aec1a5 _PyEval_EvalCodeWithName + 2981
	152 libpython3.7m.dylib                 0x0000000107ae0cf0 PyEval_EvalCode + 48
	153 libpython3.7m.dylib                 0x0000000107b258fe PyRun_FileExFlags + 174
	154 libpython3.7m.dylib                 0x0000000107b24df5 PyRun_SimpleFileExFlags + 277
	155 libpython3.7m.dylib                 0x0000000107b4831f pymain_main + 6959
	156 libpython3.7m.dylib                 0x0000000107b48efd _Py_UnixMain + 125
	157 python3.7                           0x000000010777ff24 start + 52
)
libc++abi.dylib: terminating with uncaught exception of type NSException
/nix/store/2kq3igrr75pj6nay6phchgn5x1z1f30z-stdenv-darwin/setup: line 1316: 73656 Abort trap: 6           nosetests GPy/testing/linalg_test.py GPy/testing/model_tests.py
```

**Comment by rmcgibbo** [comment]
Should I disable on Darwin?

**Comment by SuperSandro2000** [comment]
> Should I disable on Darwin?

If you have no idea why it crashes yeah.

**Comment by rmcgibbo** [comment]
There are a few hits on google for `Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Critical error: required built-in appearance SystemAppearance not found'`, but they're completely inscrutable to me. I marked it as broken on Darwin and linked to your crash report. I also pruned the tests and force-pushed.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107653` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.gpy</li>
    <li>python37Packages.gpyopt</li>
    <li>python38Packages.gpy</li>
    <li>python38Packages.gpyopt</li>
    <li>python39Packages.gpy</li>
    <li>python39Packages.gpyopt</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.paramz</li>
    <li>python38Packages.paramz</li>
    <li>python39Packages.paramz</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107653` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python37Packages.gpy</li>
    <li>python37Packages.gpyopt</li>
    <li>python37Packages.paramz</li>
    <li>python38Packages.gpy</li>
    <li>python38Packages.gpyopt</li>
    <li>python38Packages.paramz</li>
    <li>python39Packages.gpy</li>
    <li>python39Packages.gpyopt</li>
    <li>python39Packages.paramz</li>
  </ul>
</details>

**Pull request merged**